From eb0e5050a4af098b9c9dd392c324e1993b2c590e Mon Sep 17 00:00:00 2001
From: faginbagin <mythtv@hbuus.com>
Date: Mon, 10 Nov 2014 11:33:22 +0000
Subject: [PATCH] DTVRecorder: Fix 32 to 64 bit integer arithmetic in
 FindPSKeyFrames()

This is reported to fix commercial flagging of PVR-150/300/500 recordings
possibly only affecting 32 bit O/Ss due to a bad conversion from a 32 bit
unsigned int to a 64 bit signed int.

Fixes #12313.

Signed-off-by: Paul Harrison <pharrison@mythtv.org>
---
 mythtv/libs/libmythtv/recorders/dtvrecorder.cpp |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp b/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp
index ead3b33..0d2515f 100644
--- a/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp
+++ b/mythtv/libs/libmythtv/recorders/dtvrecorder.cpp
@@ -1173,7 +1173,7 @@ void DTVRecorder::FindPSKeyFrames(const uint8_t *buffer, uint len)
         if (hasKeyFrame)
         {
             _last_keyframe_seen = _frames_seen_count;
-            HandleKeyframe(_payload_buffer.size() - (bufptr - bufstart));
+            HandleKeyframe((int64_t)_payload_buffer.size() - (bufptr - bufstart));
         }
 
         if ((aspectRatio > 0) && (aspectRatio != m_videoAspect))
-- 
1.7.10.2

